ssl/statem/statem_lib.c

```diff
@@ -384,6 +384,7 @@ MSG_PROCESS_RETURN tls_process_cert_verify(SSL *s, PACKET *pkt)
     }
 #ifndef OPENSSL_NO_GOST
     {
+        pktype = EVP_PKEY_id(pkey);
```

Comment:
we don't have any strong rules.  we try to move things to an inner block if possible, but we don't create blocks *just* to limit variable scope.